### ISSUE TITLE
is there any way to use/access box id which we get from using trackers like bytetracker?

### ISSUE THREAD

**Issue by omumbare7** [ultralytics/hub#438]
### Search before asking

- [X] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and [discussions](https://github.com/ultralytics/hub/discussions) and found no similar questions.

### Question

i am experimenting of trying to do things using box ids, like lets say there are 3 detections and they get 3 unique ids, so what iam trying to do it "click picture of ID number 2 then move on to 3 then move on to 1" one by one in ascending order of ID. Is it possible to do so using yolov8 tracker?

**Comment by UltralyticsAssistant** [comment]
@omumbare7 the Ultralytics YOLOv8 tracker does not directly provide box IDs like some other trackers, such as deep_sort or bytetracker. The tracker's primary role is to track the movement of detected objects and the bounding boxes surrounding them across video frames, but it doesn't assign unique identification numbers to each object.

So in regards to your application of sequentially focusing on individual detections in ascending order by ID, I'm afraid you probably won't be able to do this directly using just the capabilities of the YOLOv8 tracker.

It is theoretically achievable with the development of custom mechanisms. This could involve concatenating frame and detection indices as pseudo-IDs for each bounding box, and then using these IDs to keep track of and manage the different detections. However, such a mechanism would need to be carefully implemented, as changes in detection or frame order, missed detections, and other factors could potentially disrupt your intended detection organization and sequence.

Remember that each YOLOv8 detection returns a tensor that includes the bounding box coordinates and class scores. Therefore, if you want to assign IDs to these detections, your custom mechanism would need to extend from this point.

Hope this helps. If you have more questions, feel free to ask.

**Comment by omumbare7** [comment]
i cannot use box id (tensor) along with the boxes it is assigned to?

**Comment by UltralyticsAssistant** [comment]
@omumbare7 the YOLOv8 tracker does not inherently output box IDs that can be used with the detected bounding boxes. The tracker is mainly designed to track the bounding boxes' movements across frames in a video. However, it isn't configured to assign a unique ID to each box.

Consequently, the use of box IDs as you're envisioning would require a customized solution. This might involve implementing your own tracking mechanism where you assign IDs to the boxes based on the frame and detection indices. However, it's important to consider that such a mechanism would need to handle various challenges, such as changes in detection order across frames, missed detections, and other similar scenarios.

I hope that clarifies your query. If you have any further questions, please let us know!

**Comment by omumbare7** [comment]
Yea that's right,  but what if the assigned ID which are to track movement of bounding boxes, does that mean bounding boxes have ID and that ID can be extracted and the corresponding box will be used?

**Comment by omumbare7** [comment]
Yea that's right,  but what if the assigned ID which are to track movement of bounding boxes, does that mean bounding boxes have ID and that ID can be extracted and the corresponding box will be used?

**Comment by UltralyticsAssistant** [comment]
@omumbare7 the YOLOv8 tracker does not assign IDs specifically to the bounding boxes. Instead, it tracks changes in bounding boxes across frames, which you could think of as implicitly assigning an 'identity' to the objects themselves (but not an unique ID). These are not extractable unique IDs, but rather an internal tracking mechanism that helps the system keep track of whether an object seen in one frame is the same object seen in subsequent frames.

So, unfortunately, you cannot extract these identities or associate them with a particular bounding box, because YOLOv8 isn't designed to assign or provide these as retrievable unique IDs. The idea of assigning unique IDs to each bounding box is beyond the scope of the tracker's function and would require a custom extension or a different kind of tracker that specifically provides this functionality.

**Comment by omumbare7** [comment]
will using deepsort help in my case?

**Comment by UltralyticsAssistant** [comment]
@omumbare7 yes, DeepSORT might be a suitable solution for you. DeepSORT is an object tracking method that combines both object detection and appearance metrics. It assigns unique identifiers to the bounding boxes that can be tracked throughout frames. Therefore, your task of focusing on a particular ID for a certain period and then moving on to the next would be made possible with DeepSORT. Just bear in mind that integrating DeepSORT into the YOLOv8 pipeline would require additional custom coding.
